src/rosdistro/manifest_provider/git.py

```diff
@@ -83,6 +83,15 @@ def git_source_manifest_provider(repo):
     return cache
 
 
+def onerror(func, path, exc_info):
+    import stat
+    if not os.access(path, os.W_OK):
+        os.chmod(path, stat.S_IWUSR)
+        func(path)
+    else:
+        raise
```

Comment:
Is there an active exception which can be reraised? Have you tried this code path?